internal/cmd/controller/reconciler/bundle_controller.go

```diff
@@ -308,3 +342,52 @@ func (r *BundleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		WithOptions(controller.Options{MaxConcurrentReconciles: r.Workers}).
 		Complete(r)
 }
+
+func (r *BundleReconciler) createDeploymentOCISecret(ctx context.Context, bundle *fleet.Bundle, bd *fleet.BundleDeployment) error {
+	namespacedName := types.NamespacedName{
+		Namespace: bundle.Namespace,
+		Name:      bundle.Spec.ContentsID,
+	}
+	var ociSecret corev1.Secret
+	if err := r.Get(ctx, namespacedName, &ociSecret); err != nil {
+		return err
+	}
+	// clone the secret, and just change the namespace so it's in the target's namespace
+	targetOCISecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ociSecret.Name,
+			Namespace: bd.Namespace,
+		},
+		Data: ociSecret.Data,
+	}
+
+	if err := controllerutil.SetControllerReference(bd, targetOCISecret, r.Scheme); err != nil {
+		return err
+	}
+	if err := r.Create(ctx, targetOCISecret); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *BundleReconciler) getOCIRegistryUrl(ctx context.Context, bundle *fleet.Bundle) (string, error) {
+	if bundle.Spec.ContentsID == "" {
+		return "", fmt.Errorf("cannot get OCI Url. Bundles's ContentsID is not set")
+	}
+	namespacedName := types.NamespacedName{
+		Namespace: bundle.Namespace,
+		Name:      bundle.Spec.ContentsID,
+	}
+	var ociSecret corev1.Secret
+	if err := r.Get(ctx, namespacedName, &ociSecret); err != nil {
+		return "", err
+	}
+	url, ok := ociSecret.Data["url"]
+	if !ok {
+		return "", fmt.Errorf("expected data [url] not found in secret: %s", bundle.Spec.ContentsID)
+	}
+	return fmt.Sprintf("oci://%s/%s:latest", string(url), bundle.Spec.ContentsID), nil
```

Comment:
We do this  a lot, but it might not work with IPv6 addresses? Is there a url module we can use?

Reply:
I'll investigate...